### ISSUE TITLE
JSX line breaks interpreted as trailing spaces

### ISSUE THREAD

**Issue by nuweb** [eslint/eslint#2443]
In routes.js (React-Flux project), having line-breaks after JSX elements gets interpreted as trailing spaces.

```
var routes = (
		<Route name="app" path="/" handler={App}>
			<Route name="dashboard" path="/app/:appId" handler={Dashboard}/>
			<DefaultRoute handler={Dashboard}/>
		</Route>
	);
```

eslint interprets them as trailing spaces,

  10:69  error  Trailing spaces not allowed  no-trailing-spaces
  11:93  error  Trailing spaces not allowed  no-trailing-spaces
✖ 2 problems (2 errors, 0 warnings)

**Comment by nzakas** [comment]
It's hard to tell from your report what exactly is being flagged. How do the errors you show map to the code you pasted? What is line 10? What is column 69?

Also, if your hypothesis that line breaks were being interpreted as trailing white space, wouldn't there be four errors instead of two?

**Comment by nuweb** [comment]
Apologies from not being descriptive enough, here's what is causing errors,

line 10  <Route name="app" path="/" handler={App}>
line 11         <Route name="dashboard" path="/app/:appId" handler={Dashboard}/>
line 12         <DefaultRoute handler={Dashboard}/>
line 13  </Route>

If I change above to code to one below, I don't get any eslint errors.

line 10		<Route name="app" path="/" handler={App}><Route name="dashboard" path="/app/:appId" handler={Dashboard}/><DefaultRoute handler={Dashboard}/>
line 11		</Route>

here's my eslint (0.19.0) configuration,

  "eslintConfig": {
    "env": {
      "browser": true,
      "node": true
    },
    "globals": {
      "ra": false,
      "require": false,
      "module": false,
      "$": false
    },
    "ecmaFeatures": {
      "jsx": true
    },
    "rules": {
      "no-console": false,
      "no-alert": false,
      "no-unused-vars": true,
      "eol-last": false,
      "no-use-before-define": 1
    }
  }

Thanks!

**Comment by nuweb** [comment]
fyi, It's not rendering correctly, so I've eliminated angle brackets in the above code sample.

**Comment by nuweb** [comment]
So, if I have below code (opening Route element on line 10 and closing Route element on line 11), it doesn't cause any lint errors

```
var routes = (
		<Route name="app" path="/" handler={App}><Route name="dashboard" path="/app/:appId" handler={Dashboard}/><DefaultRoute handler={Dashboard}/>
		</Route>
	);
```

**Comment by nuweb** [comment]
The difference that I noticed is, the elements having a preceding line-break are the elements that do not have a closing tag. A

**Comment by nuweb** [comment]
I've accidentally clicked the close button, how do I reopen the issue?

**Comment by nuweb** [comment]
One difference that I see, 

eslint doesn't complain if JSX is pure html, if JSX has React/custom components/tags, it fails to parse them correctly.

**Comment by nzakas** [comment]
What do you mean by "fail to parse"?

**Comment by nuweb** [comment]
Sorry, lint not parse.

**Comment by andersekdahl** [comment]
@nuweb If you mean that ESLint doesn't warn you if you use a React component in your JSX that you haven't defined, it's because ESLint doesn't do any linting specific to JSX/React. You'd need the eslint-plugin-react which does that.

**Comment by nuweb** [comment]
@andersekdahl 
ESLint supports JSX, which can be enabled by using config option - "jsx": true. The problem is, it's complaining about trailing white spaces preceding every nested React component (except the root component) when there are none.

**Comment by gyandeeps** [comment]
@lo1tuma Any insight on this as you are more familiar with jsx?

**Comment by lo1tuma** [comment]
@gyandeeps I can’t reproduce this with eslint 1.2.0.

**Comment by gyandeeps** [comment]
Will close.
Feel free to reopen if the issue persists.
